### DISCUSSION TITLE
Issue with Tracking Across Frames

### DISCUSSION THREAD

**Discussion by myee827** [talmolab/sleap#1463] [Help!]
Hi, I'm trying to track videos of two mice, where the white mouse is assigned to track 1 (blue) and the black mouse is assigned to track 2 (orange). I have trained SLEAP on over 1600 user-labeled frames, and it can estimate the animals' poses relatively correctly. However, when I run inference, it cannot assign the instances to the correct tracks. I have tried multiple kinds of settings for running inference, but have seen no improvement. Often, SLEAP will create multiple tracks and assign each of the animals to one of them, but none are consistent with the tracks that I want them to be assigned to. If any of the SLEAP developers could help me resolve this issue, I would very much appreciate it!! 
<img width="1440" alt="Screenshot 2023-08-21 at 10 38 59 AM" src="https://github.com/talmolab/sleap/assets/52176803/2bf7dc28-3bd5-4ce0-aa36-01d816dd44d7">
<img width="1440" alt="Screenshot 2023-08-21 at 10 36 37 AM" src="https://github.com/talmolab/sleap/assets/52176803/7f23a2f2-a201-492d-bf8f-3938dd7e32d0">
molab/sleap/assets/52176803/230d0cd3-8a9f-4e53-9382-ae72a089b03f">
<img width="1440" alt="Screenshot 2023-08-21 at 10 27 26 AM" src="https://github.com/talmolab/sleap/assets/52176803/ef297f30-f5c3-4ef8-9d6b-ac3839dcbd91">

**Comment by roomrys** [comment]
Hi @myee827,

For the non-supervised ID model, tracking is done temporally meaning SLEAP needs to "Predict On" sequential frames (whether that be an entire video or a selected clip [`Shift` + "drag on seekbar"]). The "Target Number of Instances" is good as well as the "Cull to Target Instance Count". We can also "Connect Single Track Breaks".

Let us know if that helps!

Thanks,
Liezl

P.S. For future reference, with visually distinct animals, you should be able to use supervised ID tracking which takes into account the mice's appearances. However, this would require annotating the track of each instance (select instance > Track > Add Track and then subsequently using the hotkeys [`Ctrl` +  "a number keyed to the track"](https://sleap.ai/guides/proofreading.html?highlight=assign%20track#id1)). I think we should be able to get good tracking without the supervised ID on this project.
